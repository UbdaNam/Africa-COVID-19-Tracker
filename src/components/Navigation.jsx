import { MdArrowBackIosNew, MdKeyboardVoice } from 'react-icons/md';
import { GoGear } from 'react-icons/go';
import '../styles/Navigation.css';
import { useNavigate } from 'react-router-dom';
import PropTypes from 'prop-types';

export default function Navigation({ children }) {
  const navigate = useNavigate();
  return (
    <header>
      <div className="back-button">
        <MdArrowBackIosNew
          onClick={() => navigate(-1)}
          className="header-icons"
        />
      </div>
      {children}
      <div className="header-menu">
        <MdKeyboardVoice className="header-icons" />
        <GoGear className="header-icons" />
      </div>
    </header>
  );
}

Navigation.defaultProps = {
  children: <></>,
};
Navigation.propTypes = {
  children: PropTypes.node,
};
