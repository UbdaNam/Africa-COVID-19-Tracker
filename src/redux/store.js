import { configureStore } from "@reduxjs/toolkit";
import countriesReducer from "./countries/countriesSlice";
import detailsReducer from "./details/detailsSlice";

const store = configureStore({
  reducer: {
    countries: countriesReducer,
    countryDetails: detailsReducer,
  },
});

export default store;
