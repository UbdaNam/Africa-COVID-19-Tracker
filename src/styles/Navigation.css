header {
  position: fixed;
  display: flex;
  align-items: center;
  top: 0;
  left: 0;
  width: 100%;
  padding: 0.8rem 0.3rem;
  background-color: #4369b2;
}
.back-button {
  display: flex;
  justify-content: center;
  align-items: center;
}
.header-icons {
  font-size: 1.4rem;
  color: #fafafa;
}

.header-menu {
  display: flex;
  gap: 1rem;
  margin-left: auto;
}

.header-title {
  width: 100%;
}

.header-title h5 {
  text-align: center;
  color: #f0ebeb;
  font-size: 1.1rem;
  font-weight: 400;
}
