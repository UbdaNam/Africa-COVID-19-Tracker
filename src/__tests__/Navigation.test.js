import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import store from '../redux/store';
import Navigation from '../components/Navigation';

const NavigationSnap = renderer.create(
  <Provider store={store}>
    <Router>
      <Navigation />
    </Router>
  </Provider>,
);

describe('Navigation component render correctly', () => {
  test('Check if navigation renders correctly', () => {
    expect(NavigationSnap).toMatchSnapshot();
  });
});
