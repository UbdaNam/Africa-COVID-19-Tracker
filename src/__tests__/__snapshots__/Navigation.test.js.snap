// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation component render correctly Check if navigation renders correctly 1`] = `
<header>
  <div
    className="back-button"
  >
    <svg
      className="header-icons"
      fill="currentColor"
      height="1em"
      onClick={[Function]}
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
      <path
        d="M17.77 3.77L16 2 6 12l10 10 1.77-1.77L9.54 12z"
      />
    </svg>
  </div>
  <div
    className="header-menu"
  >
    <svg
      className="header-icons"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
      <path
        d="M12 15c1.66 0 2.99-1.34 2.99-3L15 6c0-1.66-1.34-3-3-3S9 4.34 9 6v6c0 1.66 1.34 3 3 3zm5.3-3c0 3-2.54 5.1-5.3 5.1S6.7 15 6.7 12H5c0 3.42 2.72 6.23 6 6.72V22h2v-3.28c3.28-.48 6-3.3 6-6.72h-1.7z"
      />
    </svg>
    <svg
      className="header-icons"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 14 16"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M14 8.77v-1.6l-1.94-.64-.45-1.09.88-1.84-1.13-1.13-1.81.91-1.09-.45-.69-1.92h-1.6l-.63 1.94-1.11.45-1.84-.88-1.13 1.13.91 1.81-.45 1.09L0 7.23v1.59l1.94.64.45 1.09-.88 1.84 1.13 1.13 1.81-.91 1.09.45.69 1.92h1.59l.63-1.94 1.11-.45 1.84.88 1.13-1.13-.92-1.81.47-1.09L14 8.75v.02zM7 11c-1.66 0-3-1.34-3-3s1.34-3 3-3 3 1.34 3 3-1.34 3-3 3z"
        fillRule="evenodd"
      />
    </svg>
  </div>
</header>
`;
